Automattic/wp-calypso
# Purchases: Render product description for Jetpack Product purchases

Replaces #40774.

#### Changes proposed in this Pull Request

* Renders product details for Jetpack products in the manage purchase page.

> Before:
> <img width="732" alt="Screen Shot 2020-08-03 at 4 46 48 PM" src="https://user-images.githubusercontent.com/4044428/89234375-14994600-d5a9-11ea-942f-2b8ba8ecc23d.png">

> After:
> <img width="732" alt="Screen Shot 2020-08-03 at 4 46 56 PM" src="https://user-images.githubusercontent.com/4044428/89234393-1e22ae00-d5a9-11ea-8621-eaccbd5b2c46.png">


#### Testing instructions
* Navigate to `/me/purchases/` and select any Jetpack Product to view its purchase details. Ensure that the product description has been added as pictured above.
